src/excalidraw-app/data/firebase.ts

```diff
@@ -1,26 +1,44 @@
-import { getImportedKey } from "../data";
+import {
+  arrayBufferToDataURL,
+  dataURLToBlob,
+  encryptData,
+  getImportedKey,
+} from "../data";
 import { createIV } from "./index";
-import { ExcalidrawElement } from "../../element/types";
+import { ExcalidrawElement, ImageId } from "../../element/types";
 import { getSceneVersion } from "../../element";
 import Portal from "../collab/Portal";
 import { restoreElements } from "../../data/restore";
+import { BinaryFileData } from "../../types";
+import { FILE_CACHE_MAX_AGE_SEC } from "../app_constants";
 
 // private
 // -----------------------------------------------------------------------------
 
+const FIREBASE_CONFIG = JSON.parse(process.env.REACT_APP_FIREBASE_CONFIG);
+
 let firebasePromise: Promise<
   typeof import("firebase/app").default
 > | null = null;
-let firestorePromise: Promise<any> | null = null;
-let firebseStoragePromise: Promise<any> | null = null;
+let firestorePromise: Promise<any> | null | true = null;
+let firebseStoragePromise: Promise<any> | null | true = null;
+
+let isFirebaseInitialized = false;
 
 const _loadFirebase = async () => {
   const firebase = (
     await import(/* webpackChunkName: "firebase" */ "firebase/app")
   ).default;
 
-  const firebaseConfig = JSON.parse(process.env.REACT_APP_FIREBASE_CONFIG);
-  firebase.initializeApp(firebaseConfig);
+  // due to dev HMR
+  if (!isFirebaseInitialized) {
+    isFirebaseInitialized = true;
```

Comment:
we should set this to true in line 38 once its initialized

Reply:
[`03064bc` (#4011)](https://github.com/excalidraw/excalidraw/pull/4011/commits/03064bcc60cee6c0beaef2771e0c37b980777359)